### PR TITLE
Split soft source excitation for fields that evolve in Half timesteps (B, H, Hbias)

### LINKED ISSUE #87: B soft source excitation called twice right now

### PR DESCRIPTION
**Pull request by RevathiJambunathan**
B and H fields are evolved in half timesteps
FirstHalf, SecondHalf.
For these fields, the excitation is called after the evolve.
This PR ensures that the soft source excitation for such fields is added in increments of 0.5.
However, the hard source excitation is still applied fully.
Fixes #87

### CODE REVIEW

File: Source/FieldSolver/WarpXExternalEMFields.cpp [hunk 1]
```diff
@@ -154,10 +161,17 @@ WarpX::ApplyExternalFieldExcitationOnGrid (
                 WarpXUtilAlgo::getCellCoordinates(i, j, k, mfx_stag,
                                                   problo, dx, x, y, z);
                 auto flag_type = xflag_parser(x,y,z);
+                amrex::dt_type_factor = 1._rt;
+                // For soft source and FirstHalf/SecondHalf evolve
+                // the excitation is split with a prefector of 0.5
+                if (flag_type == 2._rt and dt_flag_type == 1) {
```
Comment (ajnonaka) [inline Source/FieldSolver/WarpXExternalEMFields.cpp]: all instances of `dt_flag_type` should be `dt_type_flag`

File: Source/FieldSolver/WarpXExternalEMFields.cpp [hunk 2]
```diff
@@ -154,10 +161,17 @@ WarpX::ApplyExternalFieldExcitationOnGrid (
                 WarpXUtilAlgo::getCellCoordinates(i, j, k, mfx_stag,
                                                   problo, dx, x, y, z);
                 auto flag_type = xflag_parser(x,y,z);
+                amrex::dt_type_factor = 1._rt;
```
Comment (ajnonaka) [inline Source/FieldSolver/WarpXExternalEMFields.cpp]: all instances of `amrex::dt_type_factor` should be `amrex::Real dt_type_factor`

Comment (RevathiJambunathan) [inline Source/FieldSolver/WarpXExternalEMFields.cpp]:
```suggestion
                amrex::Real dt_type_factor = 1._rt;
```

File: Source/FieldSolver/WarpXExternalEMFields.cpp (hunk 1 above)
Comment (RevathiJambunathan) [reply Source/FieldSolver/WarpXExternalEMFields.cpp]: thanks Andy. fixed!

**Review by ajnonaka — APPROVED** [review]

**Pull request merged**